int32[] desired_list
---
